pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -2,20 +2,28 @@ package ipam
 
 import (
 	"fmt"
+	"net"
 	"strings"
 
 	compute "google.golang.org/api/compute/v1"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/apimachinery/pkg/labels"
 	networkv1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1"
 	"k8s.io/klog/v2"
+	netutils "k8s.io/utils/net"
 )
 
-// PerformMultiNetworkCIDRAllocation allots pod CIDRs for all the networks that a node is connected to. It handles IPv6 only for default-network for now.
-func (ca *cloudCIDRAllocator) PerformMultiNetworkCIDRAllocation(node *v1.Node, interfaces []*compute.NetworkInterface) (defaultNwCIDRs []string, northInterfaces networkv1.NorthInterfacesAnnotation, additionalNodeNetworks networkv1.MultiNetworkAnnotation, err error) {
+// PerformMultiNetworkCIDRAllocation allots pod CIDRs for all the networks that
+// a node is connected to. It handles IPv6 only for default-network for now.
+// Any changes to API should be reflected in the 'node' variable.
+func (ca *cloudCIDRAllocator) PerformMultiNetworkCIDRAllocation(node *v1.Node, interfaces []*compute.NetworkInterface) (defaultNwCIDRs []string, err error) {
```

Comment:
This method needs a test, it has 3 nested loops with multiple API calls to different systems

Reply:
wasnt sure on how split the tests, but decided to consolidate them all in cloud_cidr_allocator_test.go, which gives full e2e testing for this function as well, see the use cases.

Reply:
I missed that sorry